irrep/__gvectors.py

```diff
@@ -62,11 +62,26 @@ def calc_gvectors(
     igp = np.zeros(3)
     igall = []
     Eg = []
+    flag = True
     for N in range(nplanemax):
         if N % 10 == 0:
+            memory = np.full(10, True)
             print(N, len(igall))
-        if len(igall) >= nplane / (2 if spinor else 1):
-            break
+        # if len(igall) >= nplane / (2 if spinor else 1):
+        #     break
+        if len(igall) >= nplane / 2:    # Only enters if vasp
+            if spinor:
+                break
+            else:      # Sure that not spinors?
+                if np.all(memory) and flag:
+                    flag = False
+                    print("WARNING: g_vectors is stuck calculating",
```

Comment:
Why is it a warning? Shouldn't it throw an exception?
Or do you think that although it is stuck, there might be no problem, and new planewaves will be found eventually? I doubt so.

Reply:
I agree to you, it makes more sense to include an exception. In principle, I think that if it enters here, it is always going to be because -spinor was set incorrectly, but I wanted to be sure.